Full-width character block (CJK) breaks rendering

The fontstack can't serve the [full-width latin characters](https://en.wikipedia.org/wiki/Halfwidth_and_fullwidth_forms) that are unfortunately common in Chinese, Japanese, and Korean (CJK), which results in entire sections of the map failing to draw when they're used. This is especially problematic when using country-localized labels (i.e., `name` and not `name_en`) in a style.

EDIT: Kanji and other CJK character sets render without any problems, only this specific block fails.

Currently, attempting to use these full-width characters throws an exception when trying to load the remote glyph PBF:

```
[ERROR] ParseTile: Parsing [8/227/99] failed: [ERROR] failed to load glyphs: HTTP status code 400; 
Raw response: {"message":"Invalid Range"}; 
URL: https://api.tiles.mapbox.com/v4/fontstack/Open%20Sans%20Regular%2c%20Arial%20Unicode%20MS%20Regular/65280-65533.pbf?access_token=pk.eyJ1IjoiZnJpZWRidW5ueSIsImEiOiJUbFBmei1zIn0.lZVQjkwWlc07RS4oEQITqg
```

[Here is an example of full-width numbers on a Japanese highway on OSM.](http://www.openstreetmap.org/edit?editor=id&relation=3793529#map=16/36.4500/137.2592) The left section of the tunnel is correct (国道41号), while the right has full-width numbers (国道４１号).

Ultimately, the fontstack needs to be able to serve full-width latin characters or MBGL should replace them with regular latin, half-width characters. #930 only temporarily fixes the drawing issues by returning a common glyph range, which means the full-width characters don't get drawn. ([Mapbox-GL-JS currently has a similar behavior.](https://friedbunny.github.io/mapbox-gl-style-dark/example.html#16.35/36.45113/-222.74053))

Cheers!

--- a/src/mbgl/text/glyph.cpp
+++ b/src/mbgl/text/glyph.cpp
@@ -7,7 +7,7 @@ GlyphRange getGlyphRange(char32_t glyph) {
     unsigned start = (glyph/256) * 256;
     unsigned end = (start + 255);
     if (start > 65280) start = 65280;
-    if (end > 65533) end = 65533;
+    if (end > 65535) end = 65535;
     return { start, end };
 }
 

